experimental/packages/opentelemetry-sdk-node/test/sdk.test.ts

```diff
@@ -582,6 +583,53 @@ describe('Node SDK', () => {
         await sdk.shutdown();
       });
 
+      describe('with unknown OTEL_NODE_RESOURCE_DETECTORS value', () => {
+        before(() => {
+          process.env.OTEL_NODE_RESOURCE_DETECTORS = 'env,os,no-such-detector';
+        });
+
+        after(() => {
+          delete process.env.OTEL_NODE_RESOURCE_DETECTORS;
+        });
+
+        // 1. If not auto-detecting resources, then NodeSDK should not
+        //    complain about `OTEL_NODE_RESOURCE_DETECTORS` values.
+        // 2. If given resourceDetectors, then NodeSDK should not complain
+        //    about `OTEL_NODE_RESOURCE_DETECTORS` values.
+        //
+        // Practically, these tests help ensure that there is no spurious
+        // diag error message when using OTEL_NODE_RESOURCE_DETECTORS with
+        // @opentelemetry/auto-instrumentations-node, which supports more values
+        // than this package (e.g. 'gcp').
+        it('does not diag.warn when not using the envvar', async () => {
```

Comment:
should this also have a test to make sure it is still sending the error message when it would? Just in case any code changes remove the message completely

just a nit, feel free to ignore it

Reply:
Fair, though I'd rather do that in a separate change... because I'd want to change its behaviour. :) Currently it is a diag.error, but I think it should be a diag.warn.

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md#enum-value says:
> if the user provides a value the implementation does not recognize, the implementation MUST generate a warning and gracefully ignore the setting.

Reply:
https://github.com/open-telemetry/opentelemetry-js/issues/4882 added for that.